# Null Hypothesis :- the defective %  not varies by centre.
# Alternate Hypothesis :- the defective %  varies by centre.

In [1]:
import pandas as pd
from scipy import stats as stats
from scipy.stats import chi2_contingency
from scipy.stats import chi2

In [2]:
Cost=pd.read_csv("Downloads\\Costomer+OrderForm.csv")
Cost.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [3]:
print(Cost['Phillippines'].value_counts(),Cost['Indonesia'].value_counts(),Cost['Malta'].value_counts(),Cost['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 Error Free    269
Defective      31
Name: Malta, dtype: int64 Error Free    280
Defective      20
Name: India, dtype: int64


In [4]:
observed=([271,267,269,280],[29,33,31,20])
observed

([271, 267, 269, 280], [29, 33, 31, 20])

In [5]:
stat,p,dof,expected = chi2_contingency([[271,267,269,280],[29,33,31,20]])

In [6]:
stat

3.858960685820355

In [7]:
p

0.2771020991233135

# statistic = 3.858960685820355,pvalue =0.2771020991233135
# As pvalue > 5%,
# Hence,we can conclude that the defective %  not varies by centre  & we fail to reject the Null Hypothesis.
            
     

     
